File: _project_review/proyecto-final-front-errorcapa8-dev/backend/src/main/java/co/icesi/TallerJPA/model/Notification.java
package co.icesi.TallerJPA.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.GenerationType;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "notifications")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String message;
    private String type;
    private boolean read;
    private LocalDateTime createdAt;
    private Long relatedEntityId;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @PrePersist
    void prePersist() {
        createdAt = LocalDateTime.now();
    }
}

File: front/bienestar-app/src/types/notification.ts
export interface UserNotification {
  id: number;
  title: string;
  message: string;
  type: string;
  read: boolean;
  createdAt: string;
  relatedEntityId?: number | null;
  userId: number;
  userName: string;
}

export interface NotificationRequest {
  title?: string;
  message: string;
  type?: string;
  relatedEntityId?: number;
  userId: number;
}
